 #!/usr/bin/env python

PACKAGE='scout_diagnostics'

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t

gen = ParameterGenerator()

#       Name                    Type      Reconfiguration level
#       Description
#       Default  Min  Max
gen.add("include_velocity_states", bool_t, 0,
        "Include velocity states in diagnostics updater.",
         True)
gen.add("include_motor_driver_states", bool_t, 0,
        "Include motor driver states in diagnostics updater.",
         False)
gen.add("include_light_states", bool_t, 0,
        "Include light states in diagnostics updater.",
         False)
gen.add("include_bms_states", bool_t, 0,
        "Include BMS states in diagnostics updater.",
         False)

exit(gen.generate(PACKAGE, "scout_diagnostics_node", "ScoutDiagnostics"))